### DISCUSSION TITLE
Feature Request: Error.ERL and stack trace properties

### DISCUSSION THREAD

**Discussion by madeinaus** [twinbasic/twinbasic#472] [Ideas]
ERL was always a hidden feature in VB, almost like MS thought that line numbers were only a hidden backward-compatibility feature, and it was never a property of the Error object.  Which made it necessary to handle ERL separately when creating a stack trace.  Which we used in our enterprise-level VBA.

So firstly, it would just be easier if ERL was also a property of the error object, and secondly, it would also be nice if there was a a subroutine name property, or a stack trace property, or a separate stack-track object.

(related note: TB handles line numbers not exactly the same as versions of VB classic and VBA I've used.)

**Comment by wqweto** [comment]
`Error` object in VBx is modelled after [`IErrorInfo` interface](https://docs.microsoft.com/en-us/windows/win32/api/oaidl/nn-oaidl-ierrorinfo) and there is no `Erl` equivalent these -- only `Description`, `Source`, `HelpFile` and `HelpContext` properties.

**Comment by WaynePhillipsEA** [comment]
I would suggest having a look at our vbWatchdog product for classic VB, since we will be incorporating vbWatchdog into tB soon.
https://www.everythingaccess.com/vbwatchdog.asp

It includes an automatic line numbering system that is relative to the procedure.

**Comment by madeinaus** [comment]
The exception object in .NET is also based on / modeled after IErrorInfo -- but it is extended with a stack trace.

:) VB predates .COM, so it's probably the case that IErrorInfo was modeled on the VB error object rather than the other way around.

**Reply by wqweto** [reply]
32-bit VB4 does not predate COM.
